### PR TITLE
Update SQL VM Rest API

### PR DESCRIPTION
**Pull request by sam0227**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by lirenhe** [comment]
@sam0227 , could you check the model validation error and fix the issue?

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 1]
```diff
@@ -1599,6 +1655,54 @@
         }
       }
     },
+    "StorageConfigurationSettings": {
+      "description": "Storage Configurations for SQL Data, Log and TempDb.",
+      "type": "object",
+      "properties": {
+        "sqlDataSettings": {
+          "$ref": "#/definitions/SQLStorageSettings",
+          "description": "SQL Server Data Stroage Settings."
```
Comment (nschonni) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]:
```suggestion
          "description": "SQL Server Data Storage Settings."
```

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 2]
```diff
@@ -1599,6 +1655,54 @@
...
+        },
+        "sqlLogSettings": {
+          "$ref": "#/definitions/SQLStorageSettings",
+          "description": "SQL Server Log Stroage Settings."
```
Comment (nschonni) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]:
```suggestion
          "description": "SQL Server Log Storage Settings."
```

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 3]
```diff
@@ -1599,6 +1655,54 @@
...
+        },
+        "sqlTempDbSettings": {
+          "$ref": "#/definitions/SQLStorageSettings",
+          "description": "SQL Server TempDb Stroage Settings."
```
Comment (nschonni) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]:
```suggestion
          "description": "SQL Server TempDb Storage Settings."
```

**Review by nschonni — COMMENTED** [review]
You could add "LUNs" to ./custom-words.txt to fix the other spelling issues

**Comment by sam0227** [comment]
@lirenhe Fixed the issues.

**Review by lirenhe — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
@sam0227 - Adding new writable properties to an existing API-version is a breaking change. It must be done in a new api-version. WHat is the impact of this property being set using for example portal, and then fetched by doing GET from an older SDK which does not understand this property (hence ignores it). And then PUT from the older SDK without this property on the same resource which has this property set.  Will the service overwrite the goal state. 
Since its preview, it may be ok, but please note that customers can still be broken or can make unintentional breaking changes to the resource.

**Comment by sam0227** [comment]
@ravbhatnagar  These properties allow the user to configure the storage pool related to SQL. If the properties are not set it'll be no-op so it should have no impact if someone do GET and PUT are from older SDK.  We don't have a plan for a new API version for now.

**Comment by sam0227** [comment]
> @sam0227 - Adding new writable properties to an existing API-version is a breaking change. It must be done in a new api-version. WHat is the impact of this property being set using for example portal, and then fetched by doing GET from an older SDK which does not understand this property (hence ignores it). And then PUT from the older SDK without this property on the same resource which has this property set. Will the service overwrite the goal state.
> Since its preview, it may be ok, but please note that customers can still be broken or can make unintentional breaking changes to the resource.

**Comment by ravbhatnagar** [comment]
SIgning off from ARM side. RP implementation is not a true PUT and so it will not cause unintentional destructive behavior for customers.

**Pull request merged**